src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,149 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
     @Override
     public void add(T value) {
+        final Node<T> newNode = new Node<>(last, value, null);
+        if (last == null) {
+            first = newNode;
+        } else {
+            last.next = newNode;
+        }
+        last = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+        } else {
+            checkIndexForAddition(index);
+            Node<T> nextNode = (index == 0) ? first : findNodeByIndex(index);
+            Node<T> previousNode = nextNode.prev;
+            Node<T> newNode = new Node<>(previousNode, value, nextNode);
+            if (previousNode == null) {
+                first = newNode;
+            } else {
+                previousNode.next = newNode;
+            }
+            nextNode.prev = newNode;
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T item : list) {
+            add(item);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return findNodeByIndex(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        Node<T> current = findNodeByIndex(index);
+        T oldValue = current.item;
+        current.item = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> nodeToRemove = findNodeByIndex(index);
+        T removedElement = nodeToRemove.item;
+        unlink(nodeToRemove);
+        return removedElement;
     }
 
     @Override
     public boolean remove(T object) {
+        for (Node<T> x = first; x != null; x = x.next) {
```

Comment:
Don't iterate your nodes from the head to the tail or vice versa. You should iterate only the necessary part of the list, depending on the index value. However, since this method is searching for a value, not an index, full iteration is required.